enterprise/javaee.wildfly/src/org/netbeans/modules/javaee/wildfly/WildflyDeploymentManager.java

```diff
@@ -83,8 +83,8 @@ public class WildflyDeploymentManager implements DeploymentManager2 {
      * server instance bcs instance properties are also removed along with
      * instance.
      */
-    private static final Map<InstanceProperties, Boolean> PROPERTIES_TO_IS_RUNNING
-            = Collections.synchronizedMap(new WeakHashMap());
+    private static final ConcurrentMap<InstanceProperties, Boolean> PROPERTIES_TO_IS_RUNNING
+            = new ConcurrentHashMap(new WeakHashMap());
```

Comment:
Hmm, isn't this very different than the original?

new ConcurrentHashMap(new WeakHashMap()) -> just creates a new ConcurrentHashMap, giving new WeakHashMap (empty map) as a parameter does nothing.

Collections.synchronizedMap(new WeakHashMap()) actually creates a synchronized version of WeakHashMap.

If ConcurrentHashMap is used, one must change the usage so that the keys inserted in the map are weak references. IF the original behavior is wanted. Right?

Reply:
You are right - would you be willing to create a PR for your suggested fix?

Reply:
> You are right - would you be willing to create a PR for your suggested fix?

I think it is faster if you guys do it. You can choose the preferred implementation also :)

Reply:
See #6872